sdk/include/reactos/svc.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * PROJECT:     ReactOS Service Host
+ * LICENSE:     BSD - See COPYING.ARM in the top level directory
+ * FILE:        sdk/include/reactos/svc.h
+ * PURPOSE:     Global Header for Service Host
+ * PROGRAMMERS: ReactOS Portable Systems Group
+ */
+/* See https://www.geoffchappell.com/studies/windows/win32/services/svchost/process/globaldata.htm?tx=78 */
+
+#ifndef __SVC_H
+#define __SVC_H
+
+#include <rpc.h>
+#include <ndk/rtlfuncs.h>
+#include <ndk/kdtypes.h>
```

Comment:
The generic rule is that shared headers for interfaces should not include "unrelated" headers.
Here, rpc.h and the ndk headers were included in svchost because the latter uses also functions and types defined in these headers for its own implementation.
However an unrelated service that implements an interface to svchost (by including this reactos/svc.h) may not need at all rpc.h and these ndk headers.
This is why, these headers may remain instead in svchost.

Reply:
Fixed. Thanks @HBelusca. :+1: You're right here too.
Moreover, the includes ordering is important too, because when I included svc.h before rpc.h, I've got the compilation error! :worried: . Only after including it **after** rpc.h, the build succeeded for me again. I wonder why it happens?... :thinking:

Reply:
Because maybe rpc.h includes itself other headers that define stuff that svc.h uses.
The usual rule is:
1. include the standard NDK + PSDK headers.
2. then include the "utilities" headers (e.g. strsafe.h, ...) and the "interfaces" headers (like that svc.h)